test/connection_switching_test.rb

```diff
@@ -208,6 +208,12 @@ def clear_connection_pool
         assert_equal 1, res.size
       end
     end
+
+    describe "when loading primary keys" do
+      it "uses the default shard" do
+        refute_nil Ticket.primary_key
+      end
+    end
```

Comment:
This test fails when we don't apply the patch to `reset_primary_key`:

```
~/Code/zendesk/active_record_shards $ TEST=test/connection_switching_test.rb SEED=64568 bundle exec rake test
/Users/colin.lancaster/Code/zendesk/active_record_shards/vendor/bundle/ruby/3.1.0/gems/activesupport-6.0.6.1/lib/active_support/core_ext/class/subclasses.rb:47: warning: method redefined; discarding old subclasses
/Users/colin.lancaster/Code/zendesk/active_record_shards/vendor/bundle/ruby/3.1.0/gems/debug-1.8.0/lib/debug/session.rb:2527: warning: assigned but unused variable - sym
/Users/colin.lancaster/Code/zendesk/active_record_shards/vendor/bundle/ruby/3.1.0/gems/debug-1.8.0/lib/debug/thread_client.rb:618: warning: assigned but unused variable - e
Run options: --seed 64568

# Running:

F..E...........................................

Finished in 12.186463s, 3.8567 runs/s, 12.4729 assertions/s.

  1) Failure:
connection switching::shard switching::when loading primary keys#test_0001_uses the default shard [/Users/colin.lancaster/Code/zendesk/active_record_shards/test/connection_switching_test.rb:214]:
Expected nil to not be nil.

  2) Error:
connection switching::shard switching::.shards.find#test_0001_works like this::
ActiveRecord::UnknownPrimaryKey: Unknown primary key for table tickets in model Ticket.
    /Users/colin.lancaster/Code/zendesk/active_record_shards/vendor/bundle/ruby/3.1.0/gems/activerecord-6.0.6.1/lib/active_record/relation/finder_methods.rb:419:in `find_with_ids'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/vendor/bundle/ruby/3.1.0/gems/activerecord-6.0.6.1/lib/active_record/relation/finder_methods.rb:69:in `find'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/shard_support.rb:26:in `block in find'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/connection_switcher.rb:58:in `block in on_all_shards'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/connection_switcher.rb:56:in `map'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/connection_switcher.rb:56:in `on_all_shards'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/shard_support.rb:9:in `each'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/lib/active_record_shards/shard_support.rb:25:in `find'
    /Users/colin.lancaster/Code/zendesk/active_record_shards/test/connection_switching_test.rb:179:in `block (4 levels) in <top (required)>'

47 runs, 152 assertions, 1 failures, 1 errors, 0 skips
```

The seed is important because the primary key could be correct if the correct shard happens to be selected when the caching happens